examples/clcomp21/Real-DEEL/Methods/base_method.py

```diff
@@ -0,0 +1,803 @@
+
+        Args:
+            setting (SettingType): Current setting from Sequoia (ClassIncremental RL or SL)
+        """ 
+        # This example is intended for classification / discrete action spaces.
+        setting.num_workers = self.hparams.num_workers
+        setting.batch_size = self.hparams.batch_size
+
+        observation_space = setting.observation_space
+        action_space = setting.action_space
+        reward_space = setting.reward_space
+        if isinstance(setting, ActiveSetting):
+            # Default batch size of 1 in RL
+            self.hparams.batch_size = 1
+            self.max_rl_train_steps = setting.steps_per_phase
+            # to enable rendering of the gym
+            self.render = False
+            self._is_rl = True
+        else:
+            # SL setting
+            assert isinstance(action_space, spaces.Discrete)
+            assert action_space == reward_space
+            self.classes_per_task = setting.increment
+        self.n_classes = action_space.n
+
+        # update model to actor critic for RL
+        model_type = model_types_map[self.hparams.model_type]
+        image_space = observation_space.x
+
+        if self._is_rl:
+            self.model = ActorCritic(model_type, image_space, action_space.n).to(
+                self.device
+            )
+        else:
+            self.model = model_type(
+                image_space, self.n_classes, bic=self.hparams.bic).to(self.device)
+
+        optim_type, optim_defaults = self._get_optim_defaults(
+            self.hparams.optimizer)
+        if "base_optim" in optim_defaults:
+            base_optim_type, base_optim_defaults = self._get_optim_defaults(
+                optim_defaults["base_optim"]
+            )
+            base_optim = base_optim_type(
+                self.model.parameters(), **base_optim_defaults,
+            )
+            optim_defaults.pop("base_optim")
+            self.optimizer = optim_type(base_optim, **optim_defaults)
+        else:
+            self.optimizer = optim_type(
+                self.model.parameters(), **optim_defaults,)
+        if self.hparams.use_scheduler:
+            scheduler_type, scheduler_defaults = schedulers_map[
+                self.hparams.scheduler_name
+            ]
+            if "step" in scheduler_defaults:
+                self._scheduler_step = scheduler_defaults["step"]
+                scheduler_defaults.pop("step")
+            self.scheduler = scheduler_type(
+                self.optimizer, **scheduler_defaults)
+
+        self.task = 0
+        self.epoch = 0
+        # smooth loss initialization
+        self.prev_loss = None
+        self.best_smooth_loss = inf
+
+        self.task_type = "RL" if self._is_rl else "SL"
+        self.iteration_name = "step" if self._is_rl else "epoch"
+        # configuration called specifically to each method inheritting base method
+        self._method_specific_configure(setting)
```

Comment:
Note: Not sure if you do it, but you could instead implement `configure` in your subclasses and do something like this:
```python3
def configure(self, setting: SettingType): 
    super().configure(setting)
    # and put the code from self._method_specific_configure() here instead

```